backend/endpoints/tests/auth_test.go

```diff
@@ -1,11 +1,93 @@
-package tests
-
-import (
-	"testing"
-)
-
-func TestLogin(t *testing.T) {
-}
-
-func TestLogout(t *testing.T) {
-}
+package tests
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"cms.csesoc.unsw.edu.au/endpoints"
+	"cms.csesoc.unsw.edu.au/endpoints/models"
+
+	"cms.csesoc.unsw.edu.au/database/repositories"
+	. "cms.csesoc.unsw.edu.au/database/repositories/mocks"
+	. "cms.csesoc.unsw.edu.au/endpoints/mocks"
+)
+
+const TEST_EMAIL = "thomas.liang1@student.unsw.edu.au"
+const TEST_PASSWORD = "backendnumberone!"
+
+// Test [endpoints.LoginHandler] succeeds on correct input.
+func TestLogin(t *testing.T) {
+	form := newTestUser()
+	responseRecorder := httptest.NewRecorder()
+	request := httptest.NewRequest("POST", "/login", nil)
+
+	// Set up mock-test boilerplate.
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	mockDependencyFactory := setUpMockRepositories(controller)
+
+	response := endpoints.LoginHandler(form, responseRecorder, request, mockDependencyFactory)
+
+	// The fun happens here!
+	const statusCode = http.StatusMovedPermanently
```

Comment:
Yeah I agree with your comments about ```http.StatusMovedPermanently``` I think it may have been added because that code was simply copy and pasted from SO? IMO our backend really shouldn't be redirecting to the dashboard page, I feel like that should be something that the frontend does instead, perhaps its best to remove that entire redirection line entirely

Reply:
I see. I'm not really sure what I should do about this.

Reply:
Hmm, would you perhaps mind if we created a separate ticket to branch off this? (Sorry for the delay btw)

Reply:
Sounds good to me.